packages/core/echo/echo-db/src/echo-handler/create.ts

```diff
@@ -40,36 +40,37 @@ export const isEchoObject = (value: unknown): value is EchoReactiveObject<any> =
   isReactiveObject(value) && getProxyHandler(value) instanceof EchoReactiveHandler;
 
 /**
- * Creates a reactive object.
+ * Creates a reactive ECHO object.
  * @internal
  */
-// TODO(burdon): Remove from public API (just use `create()`?).
-export const createObject = <T extends {}>(init: T): EchoReactiveObject<T> => {
-  invariant(!isEchoObject(init));
-  const schema = getSchema(init);
+// TODO(burdon): Document lifecycle.
+export const createObject = <T extends {}>(props: T): EchoReactiveObject<T> => {
+  invariant(!isEchoObject(props));
+  const schema = getSchema(props);
   if (schema != null) {
     validateSchema(schema);
   }
-  validateInitialProps(init);
+  validateInitialProps(props);
 
   const core = new ObjectCore();
-  if (isReactiveObject(init)) {
-    const proxy = init as any;
+  if (isReactiveObject(props)) {
+    // Already an echo-schema reactive object.
+    const proxy = props as any;
     const meta = getProxyTarget<ObjectMeta>(getMeta(proxy));
 
     // TODO(burdon): Document.
     const slot = getProxySlot(proxy);
     slot.setHandler(EchoReactiveHandler.instance);
+
     const target = slot.target as ProxyTarget;
     target[symbolInternals] = initInternals(core);
-
-    // TODO(dmaretskyi): Does this need to be disposed? RB: Probably!
-    core.updates.on(() => target[symbolInternals].signal.notifyWrite());
-
     target[symbolPath] = [];
     target[symbolNamespace] = DATA_NAMESPACE;
     slot.handler._proxyMap.set(target, proxy);
 
+    // TODO(dmaretskyi): Does this need to be disposed?
+    core.subscriptions.push(core.updates.on(() => target[symbolInternals].signal.notifyWrite()));
```

Comment:
I'd manage subscriptions on the proxy then (in ObjectInternals) -- this is only needed if we wanna drop the proxy without dropping the object-core, otherwise both objects could be garbage collected together with the callback still attached